### PR TITLE
[DOCS] wrote documentation for audio stream player 3D

### PR DESCRIPTION
**Pull request by Grosskopf**
not 100% sure about doppler tracking steps, since seems like it's not implemented jet

### CODE REVIEW

File: doc/classes/AudioStreamPlayer3D.xml [hunk 1]
```diff
@@ -131,6 +134,7 @@
 			<argument index="0" name="to_pos" type="float">
 			</argument>
 			<description>
+				Sets the position from wich audio will be played (in seconds)
```
Comment (Noshyaar) [inline doc/classes/AudioStreamPlayer3D.xml]: Missing period at the end.

**Review by NathanLovato — CHANGES_REQUESTED** [review]
Please squash or fixup with the previous commit:
`git rebase -i HEAD¬2`
Follow the instructions on screen to merge the commits together.

File: doc/classes/AudioStreamPlayer3D.xml [hunk 2]
```diff
@@ -265,67 +269,94 @@
...
+			Dampens audio above this frequency in hz.
 		</member>
 		<member name="attenuation_filter_db" type="float" setter="set_attenuation_filter_db" getter="get_attenuation_filter_db">
+			Amount in db how much the filter affects the loudness.
 		</member>
 		<member name="attenuation_model" type="int" setter="set_attenuation_model" getter="get_attenuation_model" enum="AudioStreamPlayer3D.AttenuationModel">
+			Decides if audio should get quieter with distance linearly, quadratically or logarythmically.
 		</member>
 		<member name="autoplay" type="bool" setter="set_autoplay" getter="is_autoplay_enabled">
+			If 'true' audio plays from start of the game.
```
Comment (Noshyaar) [inline doc/classes/AudioStreamPlayer3D.xml]: `If [code]true[/code], ...` with code tag and a comma between cause and effect. Also default value.

File: doc/classes/AudioStreamPlayer3D.xml [hunk 3]
```diff
@@ -265,67 +269,94 @@
...
 	</methods>
 	<members>
 		<member name="area_mask" type="int" setter="set_area_mask" getter="get_area_mask">
+			Areas in wich this sound plays.
 		</member>
 		<member name="attenuation_filter_cutoff_hz" type="float" setter="set_attenuation_filter_cutoff_hz" getter="get_attenuation_filter_cutoff_hz">
+			Dampens audio above this frequency in hz.
 		</member>
 		<member name="attenuation_filter_db" type="float" setter="set_attenuation_filter_db" getter="get_attenuation_filter_db">
+			Amount in db how much the filter affects the loudness.
```
Comment (Noshyaar) [inline doc/classes/AudioStreamPlayer3D.xml]: dB or decibels.

File: doc/classes/AudioStreamPlayer3D.xml [hunk 4]
```diff
@@ -265,67 +269,94 @@
...
 		</member>
 		<member name="attenuation_model" type="int" setter="set_attenuation_model" getter="get_attenuation_model" enum="AudioStreamPlayer3D.AttenuationModel">
+			Decides if audio should get quieter with distance linearly, quadratically or logarythmically.
```
Comment (Noshyaar) [inline doc/classes/AudioStreamPlayer3D.xml]: `logarithmically` with an 'i' instead of 'y'.

File: doc/classes/AudioStreamPlayer3D.xml [hunk 5]
```diff
@@ -265,67 +269,94 @@
...
+			Base sound level unaffected by dampening.
 		</member>
 		<member name="unit_size" type="float" setter="set_unit_size" getter="get_unit_size">
+			Factor for the attenuation effect.
 		</member>
 	</members>
 	<signals>
 		<signal name="finished">
 			<description>
+				Fires, when the audio stopped playing.
```
Comment (Noshyaar) [inline doc/classes/AudioStreamPlayer3D.xml]: Comma isn't necessary?

File: doc/classes/AudioStreamPlayer3D.xml [hunk 6]
```diff
@@ -265,67 +269,94 @@
...
 	</signals>
 	<constants>
 		<constant name="ATTENUATION_INVERSE_DISTANCE" value="0">
+			Linear dampening of loudness according to distance.
 		</constant>
 		<constant name="ATTENUATION_INVERSE_SQUARE_DISTANCE" value="1">
+			Squared dampening of loudness according to distance.
 		</constant>
 		<constant name="ATTENUATION_LOGARITHMIC" value="2">
+			Logarythmic dampening of loudness according to distance.
```
Comment (Noshyaar) [inline doc/classes/AudioStreamPlayer3D.xml]: `Logarithmic`

File: doc/classes/AudioStreamPlayer3D.xml [hunk 7]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="AudioStreamPlayer3D" inherits="Spatial" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+		Plays 3D sound in 3D space
 	</brief_description>
 	<description>
+		Plays a sound effect with directed sound effects, dampens with distance if needed, generates effect of hearable position in space.
 	</description>
 	<tutorials>
```
Comment (reduz) [inline doc/classes/AudioStreamPlayer3D.xml]: tutorials could link here:
http://docs.godotengine.org/en/latest/learning/features/audio/index.html

File: doc/classes/AudioStreamPlayer3D.xml [hunk 8]
```diff
@@ -265,67 +269,94 @@
...
+			Decides if audio should get quieter with distance linearly, quadratically or logarythmically.
 		</member>
 		<member name="autoplay" type="bool" setter="set_autoplay" getter="is_autoplay_enabled">
+			If 'true' audio plays from start of the game.
 		</member>
 		<member name="bus" type="String" setter="set_bus" getter="get_bus">
+			Bus on wich this audio is playing.
 		</member>
 		<member name="doppler_tracking" type="int" setter="set_doppler_tracking" getter="get_doppler_tracking" enum="AudioStreamPlayer3D.DopplerTracking">
+			Decides in wich step the doppler-effect should be calculated.
```
Comment (Noshyaar) [inline doc/classes/AudioStreamPlayer3D.xml]: - `which` typo
- `Doppler effect` with capitalized D and no hyphen.

File: doc/classes/AudioStreamPlayer3D.xml [hunk 9]
```diff
@@ -265,67 +269,94 @@
...
+			Amount in db how much the filter affects the loudness.
 		</member>
 		<member name="attenuation_model" type="int" setter="set_attenuation_model" getter="get_attenuation_model" enum="AudioStreamPlayer3D.AttenuationModel">
+			Decides if audio should get quieter with distance linearly, quadratically or logarythmically.
 		</member>
 		<member name="autoplay" type="bool" setter="set_autoplay" getter="is_autoplay_enabled">
+			If 'true' audio plays from start of the game.
 		</member>
 		<member name="bus" type="String" setter="set_bus" getter="get_bus">
+			Bus on wich this audio is playing.
```
Comment (Noshyaar) [inline doc/classes/AudioStreamPlayer3D.xml]: `which`

File: doc/classes/AudioStreamPlayer3D.xml (hunk 7 above)
Comment (Grosskopf) [reply doc/classes/AudioStreamPlayer3D.xml]: ok :) how should i add that? just as the Link in the next line? never saw tutorials linked in the other docs files

File: doc/classes/AudioStreamPlayer3D.xml (hunk 9 above)
Comment (Grosskopf) [reply doc/classes/AudioStreamPlayer3D.xml]: argh, i allways burn that one xD

**Comment by Noshyaar** [comment]
You can follow the guide [here](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#modifying-a-pull-request) on rebasing commits.

File: doc/classes/AudioStreamPlayer3D.xml (hunk 7 above)
Comment (NathanLovato) [reply doc/classes/AudioStreamPlayer3D.xml]: @Grosskopf As I said on Discord, place the URL inside the <tutorial> tag. It's a brand new features, fresh from reduz ^^

Comment (Grosskopf) [reply doc/classes/AudioStreamPlayer3D.xml]: thanks :D so many new channels of communication :D

**Comment by Grosskopf** [comment]
i hope i can manage this... messed up my other PR with some random commit coming in

**Comment by Grosskopf** [comment]
hooray :D

**Comment by Noshyaar** [comment]
Great job!

**Comment by Grosskopf** [comment]
thank you :)

**Pull request merged**